006-restarting-kafka-connect-connectors-and-tasks.md

```diff
@@ -0,0 +1,116 @@
+
+2. New default behaviour in the `KafkaConnector` and `KafkaMirrorMaker2` operators that automatically attempts to restart connectors and tasks if they are in the `FAILED` state. The `KafkaConnector` and `KafkaMirrorMaker2` CRs can use a new annotation to disable the automatic restarting behaviour if required.
+
+These two features could be delivered separately as they do not rely on each other.
+
+
+### `strimzi.io/restart` annotation
+
+An annotation named `strimzi.io/restart` can be applied to the `KafkaConnector` and `KafkaMirrorMaker2` custom resources. The value of the annotation and the CR it is applied to determines the connector that is restarted. Once the operator successfully calls the restart REST API endpoint, the CR is patched to remove the annotation. The operator attempts to call the restart REST API endpoint once per reconcile cycle.
+
+When applied to a `KafkaConnector` CR, the connector defined by the CR is restarted. Any value can be assigned to the annotation. For example:
+```
+apiVersion: eventstreams.ibm.com/v1alpha1
+kind: KafkaConnector
+metadata:
+  name: my-source-connector
+  annotations:
+    strimzi.io/restart: "true"
+  ...
+```
+
+When applied to a `KafkaMirrorMaker2` CR, the annotation value contains the name of the connector that is restarted. For example:
+```
+apiVersion: eventstreams.ibm.com/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mm2-cluster
+  annotations:
+    strimzi.io/restart: "source-cluster->target-cluster.MirrorSourceConnector"
+  ...
+```
+
+### `strimzi.io/restart-task` annotation
+
+An annotation named `strimzi.io/restart-task` can be applied to the `KafkaConnector`, and `KafkaMirrorMaker2` custom resources. The value of the annotation and the CR it is applied to determines the connector task that is restarted. Once the operator successfully calls the restart REST API endpoint, the CR is patched to remove the annotation. The operator attempts to call the restart REST API endpoint once per reconcile cycle.
+
+When applied to a `KafkaConnector` CR, the annotation value contains the ID of the task that is restarted. For example:
+```
+apiVersion: eventstreams.ibm.com/v1alpha1
+kind: KafkaConnector
+metadata:
+  name: my-source-connector
+  annotations:
+    strimzi.io/restart-task: "0"
+  ...
+```
+
+When applied to a `KafkaMirrorMaker2` CR, the annotation value contains the name of the connector and the ID of the task that is restarted, delimited by a `:` character. For example:
+
+```
+apiVersion: eventstreams.ibm.com/v1alpha1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mm2-cluster
+  annotations:
+    strimzi.io/restart-task: "source-cluster->target-cluster.MirrorSourceConnector:2"
```

Comment:
as for the restart connector annotation, maybe here we should have a clearer `strimzi.io/restart-connector-task`?

Reply:
👍 I have added `strimzi.io/restart-connector-tasks` for `KafkaMirrorMaker2` CRs